### PR TITLE
[Serve] Document WebSocket support in Serve

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

#35953 added experimental support for WebSockets to Serve. This change documents the feature.

Link to updated documentation: https://anyscale-ray--36735.com.readthedocs.build/en/36735/serve/http-guide.html#websockets

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - All code added for this documentation is tested in CI.

### CODE REVIEW

**Review by edoakes — APPROVED** [review]
Thank you :)

File: doc/source/serve/http-guide.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Considering your use case, you can choose the right level of abstraction:
 ## Calling Deployments via HTTP
 When you deploy a Serve application, the [ingress deployment](serve-key-concepts-ingress-deployment) (the one passed to `serve.run`) will be exposed over HTTP.
```
Comment (angelinalg) [inline doc/source/serve/http-guide.md:22]:
```suggestion
When you deploy a Serve application, the [ingress deployment](serve-key-concepts-ingress-deployment) (the one passed to `serve.run`) is exposed over HTTP.
```

**Review by angelinalg — APPROVED** [review]
Great writing!

**Comment by shrekris-anyscale** [comment]
@architkulkarni @edoakes This change is ready to merge.

**Pull request merged**